pkg/controllers/cluster/util/util.go

```diff
@@ -59,6 +59,19 @@ func GetMemberServicesForRack(ctx context.Context, r scyllav1.RackSpec, c *scyll
 	return svcList.Items, nil
 }
 
+// GetStatefulSetForRack returns rack underlying StatefulSet.
+func GetStatefulSetForRack(ctx context.Context, rack scyllav1.RackSpec, cluster *scyllav1.ScyllaCluster, kubeClient kubernetes.Interface) (*appsv1.StatefulSet, error) {
+	stsName := naming.StatefulSetNameForRack(rack, cluster)
+	sts, err := kubeClient.AppsV1().StatefulSets(cluster.Namespace).Get(ctx, stsName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
```

Comment:
Would it be less error prone to propagate the error through?

Reply:
yep it's better to handle error on caller side, fixed.